source/particles/particle.cc

```diff
@@ -220,6 +220,34 @@ namespace Particles
   }
 
 
+  template <int dim, int spacedim>
+  void
+  Particle<dim, spacedim>::update_particle_data(const void *&data)
+  {
+    const types::particle_index *id_data =
+      static_cast<const types::particle_index *>(data);
+    id                  = *id_data++;
```

Comment:
I am a bit confused. As far as I remember we are not communicating the IDs. If we do, we shouldn't because this is information which only needs to be exchanged once during setup. From my point of view, the interface here should look like the following:
```cpp
  template <int dim, int spacedim>
  void
  Particle<dim, spacedim>::update_particle_data(const numbers::particle_index id, const void *&data)
  {
    AssertDimension(id, this->id);
```
... or maybe even skip the assert totally to avoid a unnecessary data access.

Reply:
That is a good question. Right now we are still communicating the entire particle information. Even though we only use it to update the position and the properties, the data vector contains all the info about a single particle. Consequently, I preferred to keep it general (this however assumes that particle could change ID in their lifetime and etc.). However, I could remove the "id overwritten with itself".  I just felt that since we were already re-building the particle from scratch in the previous PR, this PR would just do the same without messing around with memory. This is not exactly optimal. Right now we are indeed communicating extra information that is not necessary (namely particle id and particle refence location).

Reply:
Let's keep it like this and address this in a follow-up PR.